_posts/2025-05-22-debezium-as-part-of-your-ai-solution.adoc

```diff
@@ -0,0 +1,438 @@
+intent and available candidates depending on user's query, providing
+personalized and precise service. However, user's query may be incorrect due to
+ambiguous input and typo, leading to inaccurate search. These cases may be
+released by query rewr}
+2025-05-20 06:25:00,240 INFO  [io.deb.exa.air.MilvusStore] (main) Document: {metadata={"id":"permanent"}, vector=[...], id=permanent, text=permanent}...
+----
+
+Milvus store now contains the document about IterQR framework.
+Please note that by default we cut the document to 512 characters to make demo usable even on computers without GPU.
+This behaviour is controlled by https://github.com/debezium/debezium-examples/blob/main/ai-rag/client/src/main/java/io/debezium/examples/airag/DocumentDatabase.java[DocumentDatabase] class and configuration property `debezium.rag.demo.document.truncate`.
+
+Now if the query is repeated the answer should be relevant to the real description.
+
+----
+$ java -jar target/quarkus-app/quarkus-run.jar query-chat Describe IterQR framework in 20 words
+
+2025-05-20 06:30:36,703 INFO  [io.deb.exa.air.RagCommand] (main) Sending query: Describe IterQR framework in 20 words
+2025-05-20 06:30:37,014 INFO  [io.deb.exa.air.MilvusRetrieavalAugmentor$RetrievalAugmentorDecorator] (main) Requested augmentation of UserMessage { name = null contents = [TextContent { text = "Describe IterQR framework in 20 words" }] }
+2025-05-20 06:30:44,775 INFO  [io.deb.exa.air.MilvusRetrieavalAugmentor$RetrievalAugmentorDecorator] (main) Result of augmentation is [DefaultContent { textSegment = TextSegment { text = "# Title
+IterQR: An Iterative Framework for LLM-based Query Rewrite in
+  e-Commercial Search System
+
+# Authors
+Shangyu Chen, Xinyu Jia, Yingfei Zhang, Shuai Zhang, Xiang Li, Wei Lin
+
+# Abstract
+The essence of modern e-Commercial search system lies in matching user's
+intent and available candidates depending on user's query, providing
+personalized and precise service. However, user's query may be incorrect due to
+ambiguous input and typo, leading to inaccurate search. These cases may be
+released by query rewr" metadata = {id=2504.05309v1} }, metadata = {SCORE=0.5339204333722591, EMBEDDING_ID=2504.05309v1} }, DefaultContent { textSegment = TextSegment { text = "permanent" metadata = {id=permanent} }, metadata = {SCORE=0.5046951323747635, EMBEDDING_ID=permanent} }]
+2025-05-20 06:31:08,067 INFO  [io.deb.exa.air.RagCommand] (main) Chat reply: IterQR is an iterative framework for enhancing e-Commercial search system precision, addressing user query errors via continuous iterations of candidate selection based on language model outputs.
+----
+
+From the log you can see that:
+
+* The query was augmented with the arXiv document
+* The answer corresponds to the new fact
+
+But if you query the model in *Terminal 2* again then you will receive hallucinated answer as it does not have access to augmenting facts.
```

Comment:
```suggestion
But if you query the model in *Terminal 2* again, you receive the hallucinated answer, because it does not have access to augmenting facts.
```